### PR TITLE
Add interface contact probability scores

### LINKED ISSUE #21: Request to include "contact probability" as measure of protein interaction confidence
Dear AlphaJudge team, 

I would like to request to include "contact probability" as measure of confidence for protein interaction in AlphaJudge. 
This measure was used by [Humphreys et al. 2021, Science](https://doi.org/10.1126/science.abm4805): "Using the highest AF contact probability over all residue pairs as a measure of interaction strength [...]"
At least for AF3, the contact probability does not seem to differ across the 5 models returned per seed, so it only needs to be computed once. 
Unfortunately, I have not found detailed information on this metric in the AF3 manuscript yet, and why it does not seem to differ across models.
Contact probabilities between tokens are included in the fold_[...]_full_data_[0-4].json files in the AF3 output as "contact_probs".
I have not yet confirmed whether the contact_probs matrix is strictly symmetric or not, like the PAE matrix.

Many thanks, 
Henrietta

### PR DESCRIPTION
**Pull request by DimaMolod**
Resolves #21.

## Summary
- Add contact-probability support to `Confidence` / `Interface` and write four new CSV columns:
  - `interface_contact_prob_source`
  - `interface_contact_prob_max`
  - `interface_contact_prob_top10_mean`
  - `interface_expected_contacts`
- AF3: read native `contact_probs` from `confidences.json`, align token matrices to residue matrices where possible, symmetrize, and expose as `af3_contact_probs`.
- AF2: derive contact probabilities from full result-pickle distogram logits using softmax mass for bins with upper bound `< 8 A`, matching the AlphaPulldown diagnostics fallback convention, and expose as `af2_distogram_lt_8A`.
- Keep these scores out of `interface_meta_score` for now; the benchmark comparison should decide how useful/calibrated they are.
- Add `scripts/evaluate_contact_probability_scores.py` for labelled benchmark comparison without mutating prediction directories. It can score all chain pairs, including pairs without coordinate-detected contacts, which is useful for interaction benchmarks.

## Validation
- `python -m compileall -q src/alphajudge scripts/evaluate_contact_probability_scores.py`
- `pytest -q test/test_parsers_and_runner.py::test_af2_distogram_contact_probs_softmax_cutoff_is_deterministic test/test_parsers_and_runner.py::test_contact_probability_scores_math_is_deterministic test/test_parsers_and_runner.py::test_af3_parser_accepts_alphapulldown_layout test/test_parsers_and_runner.py::test_af3_contact_probability_scores_match_raw_contact_probs` -> `4 passed in 21.38s`
- Runner smoke on copied AF2/AF3 fixtures with PNG/biophysical extras disabled:
  - AF2 fixture without result pickle: contact source blank, contact scores `nan`
  - AF3 fixture: `af3_contact_probs`, max `0.16`, top10 mean `0.146`, expected contacts `140.22`
- I also started a broader five-test run including the existing full schema runner test; the four contact-focused tests passed, and I interrupted the older broad runner path after 3:47 while it was CPU-bound in existing expensive numeric work.

## Slurm benchmark jobs
- Smoke job: `58905391`, human AF3, one pos/one neg per group, output under `/g/transform/kosinski/dima/IntAct_BioGRID_STRING/benchmark_26/contact_probability_benchmark/`.
- Full benchmark job: `58905885`, dependency `afterok:58905391`, output CSVs:
  - `/g/transform/kosinski/dima/IntAct_BioGRID_STRING/benchmark_26/contact_probability_benchmark/contact_probability_scores.csv`
  - `/g/transform/kosinski/dima/IntAct_BioGRID_STRING/benchmark_26/contact_probability_benchmark/contact_probability_metrics.csv`

### CODE REVIEW

**Comment by DimaMolod** [comment]
Pushed review-response commit 85b129d. Main updates: AF2 distogram scoring now uses the lower-bound <8A convention with logsumexp, compressed AF2 pickles are covered by tests, AF3 contact_probs alignment prefers token_res_ids and ignores extra non-residue tokens, summaries only symmetrize the requested chain-pair block, and the benchmark script now uses this checkout explicitly, complete-case metrics, tie-aware AP, checkpointing, and per-model error handling.

Benchmark status: the first full CSV in contact_probability_benchmark finished, but it imported the dirty local checkout and AF2 contact scores were present for only 12 rows, so I treat it as preliminary AF3-only signal. Corrected rerun is in /g/transform/kosinski/dima/IntAct_BioGRID_STRING/benchmark_26/contact_probability_benchmark_pr22_fix. Smoke job 58939682 completed; full Slurm job 58939710 is running.
